### PR TITLE
Patch so that `cache_size=1` is default in `torch.distributions.transforms.Transform` and elsewhere

### PR DESCRIPTION
**Pull request by stefanwebb**
This PR addresses #2076  

Essentially, when you compose a transform that requires caching for its inverse, and some of the element-wise transforms from PyTorch you can get into strife because the latter don't always do caching by default, and some don't even give you the option of setting caching (e.g. `TanhTransform` and `SigmoidTransform`)!

Here is an example of code that will raise an exception without the patches to the PyTorch transform module:

```py
import torch
import pyro.distributions as dist
import pyro.distributions.transforms as T

torch.manual_seed(0)
dim = 2
mu = torch.zeros(dim)
log_std = torch.zeros(dim)
transforms = [T.Sylvester(input_dim=dim), T.TanhTransform()]
d = dist.TransformedDistribution(dist.Normal(mu, log_std.exp()), transforms)

y = d.sample()
print(d.log_prob(y))

```

I add the discussion label because I just want to run past @fritzo, do you think it's okay to set `Transform` to do caching by default, rather than adding a `cache_size=1` keyword argument to every element-wise transform?

### CODE REVIEW

**Comment by stefanwebb** [comment]
Ah I see! Let me do a mock up of option (2) to see if that would work.

Yes I think all transforms that don't require caching should give you the option at least of caching them in a keyword arg to `__init__` (with default value 0) so will PR that to PyTorch

**Comment by fritzo** [comment]
@stefanwebb I needed the `ComposeTransform` fix to solve another problem. Can we collaborate on fixing this upstream in Pytorch https://github.com/probtorch/pytorch/pull/153 ?

**Comment by stefanwebb** [comment]
I'm going to close this because it's superseded by https://github.com/probtorch/pytorch/pull/153
